src/masternodes/validation.cpp

```diff
@@ -2400,7 +2402,182 @@ static Res ValidateCoinbaseXVMOutput(const XVM &xvm, const FinalizeBlockCompleti
     return Res::Ok();
 }
 
-static Res ProcessEVMQueue(const CBlock &block, const CBlockIndex *pindex, CCustomCSView &cache, const CChainParams& chainparams, const uint64_t evmQueueId) {
+CTransferDomainMessage DecodeTransferDomainMessage(const CTransactionRef& tx, const CBlockIndex* pindex, const CChainParams& chainparams) {
+    std::vector<unsigned char> metadata;
+    CustomTxType txType = GuessCustomTxType(*tx, metadata);
+    if (txType == CustomTxType::TransferDomain) {
+        auto txMessage = customTypeToMessage(txType);
+        if (CustomMetadataParse(pindex->nHeight, chainparams.GetConsensus(), metadata, txMessage))
+            return *(std::get_if<CTransferDomainMessage>(&txMessage));
+    }
+    return CTransferDomainMessage{};
+}
+
+CAmount GetEvmDST20TotalSupply(const DCT_ID& id, const std::string &stateRoot = std::string()) {
```

Comment:
Would it be safer to have this as `ResVal<CAmount>`?

Reply:
Correct, switched to ResVal. Thnx